### PR TITLE
gosu 1.11 - no build.sh

### PR DESCRIPTION
**Pull request by sodre**
@jakirkham,

   I think the go recipes are becoming more and more succinct as we move along. There are still some things that could be better in the long-run, but the only way to hone them is pack more go packages. 

Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Closes https://github.com/conda-forge/gosu-feedstock/pull/5

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,35 +1,48 @@
-{% set version = "1.10" %}
-{% set sha256 = "bcbe8ada8c435a253b1fa438f505dd311832a88f52316fd7d30342b106a4242c" %}
-{% set runc_version = "0.1.0" %}
-{% set runc_sha256 = "78ae42905c0f26fcfa9353e48e1a001b5b5c3c94b33e2ceacc6bc84c23c61a88" %}
+{% set goname = "github.com/tianon/gosu" %}
+{% set version = "1.11" %}
```
Comment (sodre) [inline recipe/meta.yaml:2]: Most recipes would only need those two lines.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,35 +1,48 @@
...
+
+{% set name = goname.split('/')[-1] %}
+{% set pkg_src = ('src/'+goname).replace("/",os.sep) %}
```
Comment (sodre) [inline recipe/meta.yaml]: These helps to keep the recipe DRY, from the goname and version we can pickup the target directory and the short name of the program.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,35 +1,48 @@
...
+
+# The runc version much match the upstream Dockerfile
+# https://github.com/tianon/gosu/blob/master/Dockerfile
+{% set runc_goname = "github.com/opencontainers/runc" %}
+{% set runc_version = "1.0.0-rc5" %}
+{% set runc_pkg_src = ('src/'+runc_goname).replace("/",os.sep) %}
```
Comment (sodre) [inline recipe/meta.yaml]: gosu does things in a  strange way. I expect this to change in the future when `go modules` catches on. This was one of the main new features of go 1.11.

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,35 +1,48 @@
...
 
 package:
-  name: gosu
+  name: {{ name }}
```
Comment (sodre) [inline recipe/meta.yaml:14]: My suggestion to have the template recipe have the name set to go-{{ name }}, following the r-{{ name }} template. However, this is a legacy package and we are better off not changing things.

Another option, is to create a metapackage with he canonical go/conda package.

File: recipe/meta.yaml [hunk 5]
```diff
@@ -1,35 +1,48 @@
...
+    sha256: 770e180d1cca1f3c1a67512ebf9720caafe97af4fc2fe12fa43a3cf7bfa77522
 
 build:
-  number: 1001
+  number: 1000
   skip: true  # [not linux]
+  script:
+    - pushd {{ pkg_src }}
+    - go install -v -ldflags '-d -s -w'
+    - go test -v
```
Comment (sodre) [inline recipe/meta.yaml:32]: Unfortunately, we need the entire source code if we want to test. This is similar to our calls to `make check` for c/c++ programs.

File: recipe/conda_build_config.yaml [hunk 6]
```diff
@@ -0,0 +1,3 @@
+go_compiler:
+  - go-nocgo
```
Comment (sodre) [inline recipe/conda_build_config.yaml:2]: I expect this to be redundant when I finish the work on conda-forge/conda-forge-pinning-feedstock#169.

**Review by sodre — COMMENTED** [review]
I left some comments to help you walk-through my changes.

**Comment by sodre** [comment]
@conda-forge-admin please rerender

File: recipe/meta.yaml (hunk 2 above)
Comment (sodre) [inline recipe/meta.yaml]: Also, the `os.sep` call in L5 is what causing the linter to fail. I already submitted a PR to conda-forge/conda-smithy#980 which was merged but is pending a release.

Comment (sodre) [reply recipe/meta.yaml]: in the meantime to get the error out I replaced it with an inline if statement.

**Comment by sodre** [comment]
@conda-forge/gosu this is ready for review and possibly merge.

Also, we should be able to build this for different architectures if needed.

File: recipe/meta.yaml [hunk 7]
```diff
@@ -1,35 +1,48 @@
...
 
 requirements:
   build:
     - {{ compiler('go') }}
 
 test:
   commands:
-    - which gosu
```
Comment (jakirkham) [inline recipe/meta.yaml]: Does `which` still work?

Comment (sodre) [reply recipe/meta.yaml]: I am not sure I understand what you mean. Was it removed recently?

File: recipe/meta.yaml [hunk 8]
```diff
@@ -1,35 +1,48 @@
-{% set version = "1.10" %}
-{% set sha256 = "bcbe8ada8c435a253b1fa438f505dd311832a88f52316fd7d30342b106a4242c" %}
-{% set runc_version = "0.1.0" %}
-{% set runc_sha256 = "78ae42905c0f26fcfa9353e48e1a001b5b5c3c94b33e2ceacc6bc84c23c61a88" %}
+{% set goname = "github.com/tianon/gosu" %}
+{% set version = "1.11" %}
+
+{% set name = goname.split('/')[-1] %}
+{% set pkg_src = ('src/'+goname).replace("/","\\" if win else "/") %}
```
Comment (jakirkham) [inline recipe/meta.yaml]: Should we worry about Windows given this is Linux only?

File: recipe/meta.yaml [hunk 9]
```diff
@@ -1,35 +1,48 @@
...
+
+# The runc version much match the upstream Dockerfile
+# https://github.com/tianon/gosu/blob/master/Dockerfile
+{% set runc_goname = "github.com/opencontainers/runc" %}
+{% set runc_version = "1.0.0-rc5" %}
+{% set runc_pkg_src = ('src/'+runc_goname).replace("/", "\\" if win else "/") %}
```
Comment (jakirkham) [inline recipe/meta.yaml]: Same question as above.

File: recipe/meta.yaml (hunk 8 above)
Comment (sodre) [reply recipe/meta.yaml]: The recipe was based on the template we talked about late last year. I have no problem removing it, or we can just keep it in for consistency.

File: recipe/meta.yaml (hunk 7 above)
Comment (sodre) [reply recipe/meta.yaml]: done.

File: recipe/meta.yaml (hunk 8 above)
Comment (jakirkham) [reply recipe/meta.yaml]: Sure, whatever is convenient for you.

File: recipe/meta.yaml [hunk 10]
```diff
@@ -1,24 +1,35 @@
...
+    url: https://{{ runc_goname }}/archive/v{{ runc_version }}.tar.gz
+    sha256: 770e180d1cca1f3c1a67512ebf9720caafe97af4fc2fe12fa43a3cf7bfa77522
 
 build:
-  number: 1001
+  number: 1000
   skip: true  # [not linux]
+  script:
+    - pushd {{ pkg_src }}
+    - go install -v -ldflags '-d -s -w'
```
Comment (jakirkham) [inline recipe/meta.yaml:31]: Should we cleanup the Go package cache? Does that wind up in the `PREFIX`?

**Comment by jakirkham** [comment]
Thanks @sodre. Couple questions above.

File: recipe/meta.yaml (hunk 10 above)
Comment (sodre) [reply recipe/meta.yaml:31]: No need to clean-up the cache with our current CI setup. The go package cache is based on the GOPATH variable which is set to $SRC_DIR. 

I built the package locally and here is list of the files to verify:
```
" tar.vim version v29
" Browsing tarfile /opt/conda/conda-bld/linux-64/gosu-1.11-h375a9b1_1000.tar.bz2
" Select a file with cursor and press ENTER

info/files
info/hash_input.json
info/paths.json
info/index.json
info/about.json
info/LICENSE.txt
info/git
info/test/run_test.sh
info/recipe/conda_build_config.yaml
info/recipe/meta.yaml
bin/gosu
info/recipe/meta.yaml.template
```

**Review by jakirkham — APPROVED** [review]

File: recipe/meta.yaml (hunk 10 above)
Comment (jakirkham) [reply recipe/meta.yaml:31]: Ah ok. Thanks for the reminder. Looks good.

Comment (sodre) [reply recipe/meta.yaml:31]: 👍

**Comment by sodre** [comment]
@jakirkham, this is ready to merge.

**Comment by jakirkham** [comment]
Thanks @sodre :)

**Pull request merged**